### ISSUE TITLE
[Federation] few issues in federation code identified by go vet and staticcheck

### ISSUE THREAD

**Issue by shashidharatd** [kubernetes/kubernetes#42172]
Following issues are in the latest code in the repo.
go vet issues:
```
#~/gopath/src/k8s.io/kubernetes/federation# go vet ./...
client/clientset_generated/federation_clientset/fake/clientset_generated.go:56: literal copies lock value from fakePtr: k8s.io/kubernetes/vendor/k8s.io/client-go/testing.Fake
exit status 1
client/clientset_generated/federation_internalclientset/fake/clientset_generated.go:56: literal copies lock value from fakePtr: k8s.io/kubernetes/vendor/k8s.io/client-go/testing.Fake
exit status 1
```
staticcheck issues:
```
#~/gopath/src/k8s.io/kubernetes/federation# staticcheck ./...
pkg/federation-controller/util/delaying_deliverer.go:124:4: ineffective break statement. Did you mean to break out of the outer loop? (SA4011)
pkg/federation-controller/replicaset/replicasetcontroller.go:544:13: this value of err is never used (SA4006)
pkg/federation-controller/deployment/deploymentcontroller.go:530:13: this value of err is never used (SA4006)
pkg/dnsprovider/providers/google/clouddns/clouddns.go:83:4: oauth2.NoContext is deprecated: Use context.Background() or context.TODO() instead.  (SA1019)
pkg/dnsprovider/providers/google/clouddns/clouddns.go:94:34: oauth2.NoContext is deprecated: Use context.Background() or context.TODO() instead.  (SA1019)
pkg/dnsprovider/providers/aws/route53/route53.go:63:21: session.New is deprecated: Use NewSession functions to create sessions instead. NewSession has the same functionality as New except an error can be returned when the func is called instead of waiting to receive an error until a request is made.  (SA1019)
pkg/kubefed/init/init.go:442:3: api.NodeLegacyHostIP is deprecated: NodeLegacyHostIP will be removed in 1.7.  (SA1019)
pkg/kubefed/init/init_test.go:1285:4: the surrounding loop is unconditionally terminated (SA4004)
pkg/federation-controller/service/servicecontroller.go:789:3: this value of servicesToUpdate is never used (SA4006)
pkg/federation-controller/service/servicecontroller.go:972:3: this value of err is never used (SA4006)
```
cc @kubernetes/sig-federation-bugs

**Comment by MrHohn** [comment]
@shashidharatd Sorry for commenting on this old issue. What was the resolution of the `go vet` error? Hit a similar problem on https://github.com/kubernetes/sample-controller/issues/16.

**Comment by shashidharatd** [comment]
@MrHohn, We didn't fix the issues in federation as it was generated code.

**Comment by MrHohn** [comment]
@shashidharatd Thanks for your reply. Yeah I found someone is attempting to fix that on kubernetes/kubernetes#60584.
